byoc/byoc.go

```diff
@@ -46,38 +37,141 @@ func NewBYOCGatewayServer(node *core.LivepeerNode, statusStore StatusStore, whip
 		sharedBalMtx: &sync.Mutex{},
 	}
 
+	bsg.StreamPipelines = make(map[string]*BYOCStreamPipeline)
+
 	bsg.registerRoutes()
 	return bsg
 }
 
-func NewBYOCOrchestratorServer(node *core.LivepeerNode, orch Orchestrator, trickleSrv *trickle.Server, trickleBasePath string, mux *http.ServeMux) *BYOCOrchestratorServer {
-	bso := &BYOCOrchestratorServer{
-		node:            node,
-		orch:            orch,
-		trickleSrv:      trickleSrv,
-		trickleBasePath: trickleBasePath,
-		httpMux:         mux,
-		sharedBalMtx:    &sync.Mutex{},
-	}
-
-	bso.registerRoutes()
-	return bso
-}
-
 func (bsg *BYOCGatewayServer) Node() *core.LivepeerNode {
 	return bsg.node
 }
 func (bsg *BYOCGatewayServer) SharedBalanceLock() *sync.Mutex {
 	return bsg.sharedBalMtx
 }
 
+func (bsg *BYOCGatewayServer) newStreamPipeline(requestID, streamID, pipeline string, streamParams byocAIRequestParams, streamRequest []byte) *BYOCStreamPipeline {
+	streamCtx, streamCancel := context.WithCancelCause(context.Background())
+	bsg.mu.Lock()
+	defer bsg.mu.Unlock()
+
+	//ensure streamRequest is not nil or empty to avoid json unmarshal issues on Orchestrator failover
+	//sends the request bytes to next Orchestrator
+	if streamRequest == nil || len(streamRequest) == 0 {
```

Comment:
Small nit but wouldn't this work given that len of nil byte gives `true`? Not to important but just something I noticed.
```suggestion
// Ensure a non-empty request body for failover
if len(streamRequest) == 0 {
	streamRequest = []byte("{}")
}
```

Reply:
Feel free to ignore this one very minor. Just something I noticed.